### DISCUSSION TITLE
[Bug] `scoop add bucket` does not react correctly with git `early EOF` error

### DISCUSSION THREAD

**Discussion by LuoQuestionmark** [ScoopInstaller/Scoop#5884] [Ideas]
#### Current Behavior

While typing `scoop add bucket <bucket>`, the searching process can fail with git `early EOF` error. While the searching/sync process is failed, the error is not handled correctly with the scoop.

```Powershell
PS C:\Users\WenhaoLUO> scoop bucket add extras
Checking repo... error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 3749 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
OK
The extras bucket was added successfully.
```

Furthermore, after the wrongly handled error,

```Powershell
PS C:\Users\WenhaoLUO> scoop bucket add extras
WARN  The 'extras' bucket already exists. To add this bucket again, first remove it by running 'scoop bucket rm extras'.
PS C:\Users\WenhaoLUO> scoop bucket add extras
WARN  The 'extras' bucket already exists. To add this bucket again, first remove it by running 'scoop bucket rm extras'.
PS C:\Users\WenhaoLUO> scoop update
Updating Scoop...
Updating 'extras' bucket...
'extras' is not a git repository. Skipped.
```

#### Expected Behavior

While the git behave with an error, scoop should handle it correctly by reporting the `bucket add` command is not executed correctly.

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:**

```txt
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      22621  2506
```

**Additional software:** 

```txt
git version 2.44.0.windows.1
```

#### Scoop Configuration

```json
{
    "last_update":  "2024-02-27T09:02:34.8054749+01:00",
    "scoop_branch":  "master",
    "scoop_repo":  "https://github.com/ScoopInstaller/Scoop"
}
```

**Comment by rashil2000** [comment]
How can I repro this error?

**Reply by LuoQuestionmark** [reply]
Emm... A bad internet connection and an installation of `scoop` without buckets such as "extra" should do the work.

If you cannot find an internet connection that is bad enough, you can configure git as the follow in `.gitconfig` (c.f. [this answer](https://stackoverflow.com/a/29355320) on stack overflow).

```config
[core]
packedGitLimit = 1m
packedGitWindowSize = 1m
[pack]
deltaCacheSize = 1m
packSizeLimit = 1m
windowMemory = 1m
```

Then you should get a clean installation of `scoop`, after that you can try to add buckets as with the command `scoop bucket add extras`.

P.S. Since these operations can have a global impact on your git and on your system, you should backup the related folders.
